src/components/JsonViewer/JsonViewer.tsx

```diff
@@ -93,26 +89,25 @@ class Json extends React.PureComponent<JsonProps> {
     }
   };
 
-  clickListener = (event: MouseEvent) => {
-    this.collapseElement(event.target as HTMLElement);
-  };
+  const clickListener = React.useCallback((event: MouseEvent) => {
+    collapseElement(event.target as HTMLElement);
+  }, []);
 
-  focusListener = (event: KeyboardEvent) => {
+  const focusListener = React.useCallback((event: KeyboardEvent) => {
     if (event.key === 'Enter') {
-      this.collapseElement(event.target as HTMLElement);
+      collapseElement(event.target as HTMLElement);
     }
-  };
+  }, []);
 
-  componentDidMount() {
-    this.node!.addEventListener('click', this.clickListener);
-    this.node!.addEventListener('focus', this.focusListener);
-  }
+  React.useEffect(() => {
+    if (node) {
+      node?.addEventListener('click', clickListener);
+      node?.addEventListener('focus', focusListener);
+    }
+  }, [clickListener, focusListener, node]);
```

Comment:
this hook will be running every time component updates, is it ok?

if no, u can add useMount hook

```js
export const useMount = (callback: () => void): void => {
  useEffect(() => {
    callback();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
};
```

Reply:
I think yes. It does not work correctly with useMount hook.

Reply:
yeah, probably because on initial render we don't have a node